docs/usage-inventory-catalog.md

```diff
@@ -307,9 +307,45 @@ Once you run `anta nrfu table`, you will see following output:
 └───────────┴────────────────────────────┴─────────────┴────────────┴───────────────────────────────────────────────┴───────────────┘
 ```
 
-### Example script to merge catalogs
+### Merging Catalogs
 
-The following script reads all the files in `intended/test_catalogs/` with names `<device_name>-catalog.yml` and merge them together inside one big catalog `anta-catalog.yml`.
+**Example script to merge catalogs using the new `merge_catalogs` function**
+
+The following script reads all the files in `intended/test_catalogs/` with names `<device_name>-catalog.yml` and merge them together inside one big catalog `anta-catalog.yml` using the new `merge_catalogs` function.
+
+```python
+#!/usr/bin/env python
+from anta.catalog import AntaCatalog, merge_catalogs
+
+from pathlib import Path
+from anta.models import AntaTest
+
+
+CATALOG_SUFFIX = "-catalog.yml"
+CATALOG_DIR = "intended/test_catalogs/"
+
+if __name__ == "__main__":
+    catalogs = []
+    for file in Path(CATALOG_DIR).glob("*" + CATALOG_SUFFIX):
+        device = str(file).removesuffix(CATALOG_SUFFIX).removeprefix(CATALOG_DIR)
+        print(f"Loading test catalog for device {device}")
+        catalog = AntaCatalog.parse(file)
+        # Add the device name as a tag to all tests in the catalog
+        for test in catalog.tests:
+            test.inputs.filters = AntaTest.Input.Filters(tags={device})
+        catalogs.append(catalog)
+
+    # Merge all catalogs
+    merged_catalog = merge_catalogs(catalogs)
+
+    # Save the merged catalog to a file
+    with open(Path('anta-catalog.yml'), "w") as f:
+        f.write(catalog.dump().yaml())
+```
+!!! warning
+    The `AntaCatalog.merge()` method is deprecated and will be removed in ANTA v2.0. Please use the `merge_catalogs()` function from the `anta.catalog` module instead.
+
+**For reference: Deprecated method (to be removed in ANTA v2.0)**
```

Comment:
do we need to keep the deprecated method as an example? I would remove it to avoid maintaining this and have cleaner doc - keep the warning though

Reply:
I'm also fine removing it. Fixed in 28e3fa985af88c21c1403185c5a07eb1cdd61a24